source/bao_hyp/config.rst

```diff
@@ -68,36 +68,280 @@ relative to the config source file.
+    };
+
+where:
+
+- **pa** [mandatory] - corresponds to the base physical address of the device;
+- **va** [mandatory] - corresponds to the base virtual address of the device;
+- **size** [mandatory] - corresponds to the size of the device memory region;
+- **interrupt_num** [optional] - corresponds to the number of interrupts generated by the device to
+  the VM. By default, ``interrupt_num`` equals to 0;
+- **interrupts** [mandatory if *interrupt_num*>0] - defines a list of interrupt IDs generated by
+  the device - ``(irqid_t[]) {irq_1, ..., irq_n};``
+- **id** [optional] - corresponds to the bus master id for iommu effects:
+
+.. warning::
+  Specifying a number of interrupts in the ``interrupts`` buffer that differs from the
+  ``interrupt_num`` may result in unforeseen behavior.
 
 5. Memory Management
 ####################
+.. _Memory Management:
+
+ **mmu** [optional] - In MPU-based platforms which might also support virtual memory (i.e. aarch64
+ cortex-r) the hypervisor sets up the VM using an MPU by default. If the user wants this VM to use
+ the MMU they must set the config ``mmu`` parameter to true;
 
 6. Architectural-Specific Configurations
 ########################################
+.. _Architectural-Specific Configurations:
+
+- **arch** [mandatory] - allows the definition of architecture dependent configurations and is
+  configured through the following structure:
+
+.. code-block:: c
+
+    struct arch_vm_platform {
+
+        // Configuration of the Generic Interrupt Controller (GIC)
+        struct vgic_dscrp {
+            paddr_t gicd_addr;
+            paddr_t gicc_addr;
+            paddr_t gicr_addr;
+            size_t interrupt_num;
+        } gic;
+
+        // Configuration of the System Memory Management Unit (SMMU)
+        struct {
+            streamid_t global_mask;
+            size_t group_num;
+            struct smmu_group {
+                streamid_t mask;
+                streamid_t id;
+            } *groups;
+        } smmu;
+    };
+
+
+where:
+
+- **vgic_dscrp** [mandatory] - corresponds to the configuration of the Generic Interrupt Controller
+  (GIC);
+- **vgic_dscrp** [optional] - corresponds to the configuration of the System Memory Management Unit
+  (SMMU);
+
 
 CPU Affinity
 ************
 
+The configuration file of the Bao hypervisor also enables the definition of core affinity, which
+involves selecting the physical core where the guest should run.
+
+.. figure:: img/cpu-affinity.svg
+    :align: center
+    :name: cpu-affinity-fig
+
+This functionality is achieved through the following configuration parameter:
+
+- **cpu_affinity** [optional] - corresponds to a bitmap signaling the preferred physical CPUs
+  assigned to the VM. If this value is mutually exclusive for all the VMs, the physical CPUs
+  assigned to each VM follow the bitmap. Otherwise (in case of bit overlap or lack of affinity
+  definition), the CPU assignment is defined by the hypervisor;
```

Comment:
the cpu affinity will be as best as possible be followed, but it is not guaranteed.

Reply:
Thanks, added this information!